lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -181,6 +181,189 @@
+            }
+            position = 0;
+            lastSize = bucketSize;
+            bucketSize *= 2;
+        }
+    });
+
+    platform.registerChakraLibraryFunction("DefaultSortCompare", function(left, right) {
+        // as the sort only ever checks for < 0 on the return value of compare functions
+        // only have to handle this case
+        if (left.string < right.string) {
+            return -1;
+        }
+        return 0;
+    });
+
+    platform.registerChakraLibraryFunction("CreateCompareArray", function(array, length) {
+        const compArray = [];
+        compArray.__proto__ = null;
+        let i = 0;
+        let value;
+        while (i < length) {
+            value = array[i];
+            compArray[i++] = {
+                value : value,
+                string : "" + value
+            };
+        }
+        return compArray;
+    });
+
+    platform.registerChakraLibraryFunction("FillArrayHoles", function(array, length, offset) {
+        let i = offset, j = offset, holes = 0;
+        let value;
+        while (i < length) {
+            value = array[i];
+            if (value !== undefined) {
+                array[j++] = value;
+            } else if (!(i in array)) {
+                ++holes;
+            }
+            ++i;
+        }
+
+        const valuesLength = j;
+        const hasLength = length - holes;
+        while (j < hasLength) {
+            array[j++] = undefined;
+        }
+        while (j < length) {
+            delete array[j++];
+        }
+        return valuesLength;
+    });
+
+    platform.registerFunction(platform.FunctionKind.Array_sort, function (compareFn) {
+        //#sec-array.prototype.sort
+        let defaultSort = false;
+        if (compareFn === undefined) {
+            defaultSort = true;
+            compareFn = __chakraLibrary.DefaultSortCompare;
+        } else if (typeof compareFn !== "function") {
+            __chakraLibrary.raiseFunctionArgument_NeedFunction("Array.prototype.sort");
+        }
+
+        const {o, len} = __chakraLibrary.CheckArrayAndGetLen(this, "Array.prototype.sort");
+
+        if (len < 2) { // early return if length < 2
+            return o;
+        }
+
+        // check for if the array has any missing values
+        // also pull in any values from the prototype
+        let i = 0, length = len;
+        while (i < len) {
+            if (o[i] === undefined) {
+                length = __chakraLibrary.FillArrayHoles(o, len, i);
+                break;
+            }
+            o[i] = o[i++];
+        }
+
+        let compArray = o;
+        if (defaultSort === true && length > 1) {
+            compArray = __chakraLibrary.CreateCompareArray(o, length);
+        }
+
+        // for short arrays perform an insertion sort
+        if (length < 2048) {
+            i = 1;
+            let j = 0, k = 0, l = 0;
+            while (i < length) {
+                const item = compArray[i];
+                l = i;
```

Comment:
I understand that l comes after ijk, but it's also really hard to read. Could you use a different name please?

Reply:
In general, I would prefer more descriptive names too, like i->numSorted, j->lowerBound, k->insertionPoint, l->upperBound.

---
In reply to: [221307258](https://github.com/Microsoft/ChakraCore/pull/5724#discussion_r221307258) [](ancestors = 221307258)